processing/src/main/java/io/druid/query/aggregation/JavaScriptAggregatorFactory.java

```diff
@@ -300,13 +299,9 @@ public String toString()
            '}';
   }
 
-  private JavaScriptAggregator.ScriptAggregator getCompiledScript()
+  private void checkCompiledScript()
   {
-    if (compiledScript == null) {
-      throw new ISE("JavaScript is disabled");
-    }
-
-    return compiledScript;
+    compiledScript = compiledScript == null ? compileScript(fnAggregate, fnReset, fnCombine) : compiledScript;
```

Comment:
Is this going to be thread-safe (albeit with some possible extra work if multiple threads don't see the `compiledScript` variable update)? AggregatorFactorys definitely do need to be thread-safe as they are used from multiple threads during query processing.

Reply:
Thanks. Fixed.